src/orchard/xref.clj

```diff
@@ -57,13 +57,18 @@
   {:added "0.5"}
   [v]
   (when-let [^clojure.lang.AFn v (to-fn v)]
-    (let [f-class-name (-> v .getClass .getName)]
-      ;; this uses the implementation detail that the clojure compiler always
-      ;; prefixes names of lambdas with the name of its surrounding function class
-      (into #{} (comp (filter (fn [[k _v]] (clojure.string/includes? k f-class-name)))
-                      (map (fn [[_k value]] (.get ^java.lang.ref.Reference value)))
-                      (mapcat fn-deps-class))
-            class-cache))))
+    (let [f-class-name (-> v .getClass .getName)
+          ;; this uses the implementation detail that the clojure compiler always
+          ;; prefixes names of lambdas with the name of its surrounding function class
+          deps (into #{} (comp (filter (fn [[k _v]] (clojure.string/includes? k f-class-name)))
+                               (map (fn [[_k value]] (.get ^java.lang.ref.Reference value)))
+                               (mapcat fn-deps-class))
+                     class-cache)]
+      ;; if there's no deps the class is most likely AoT compiled,
+      ;; try to access it directly
+      (if (empty? deps)
```

Comment:
What if it's just a defn with no deps?

e.g. some new code I'm just writing now.

Reply:
then fn-deps-class will still return an empty set, we just waste some time by calling it twice.
the other option would be to run the filter outside the transducer to check if it returns empty. that would mean we don't have to run fn-deps-class twice for empty functions.
In the end, running fn-deps-class twice on empty functions doesn't really make much of a difference since it's very fast, especially when it doesn't have anything to return.

Reply:
Alright then!